src/modes.cpp

```diff
@@ -268,6 +295,20 @@ void aste::runMapperMode(const aste::ExecutionContext &context, const OptionMap
           }
           VLOG(1) << "Data written: " << asteInterface.mesh.previewData(meshdata);
         }
+#ifdef ASTE_NN_GRADIENT_MAPPING
```

Comment:
Can we also add a preview for the gradient data?

Reply:
Maybe even directionwise, i.e., derivative in x-direction, y-direction... Would be a nice sanity check for the user.

Reply:
Yes we can add previre but the data is not written direction-wise.

Always follows `dx dy dz`  for scalar and `c1dx c2dx c1dy c2dy....` as the vector given to preCICE interface